lib/model/folder.go

```diff
@@ -439,6 +440,7 @@ func (f *folder) scanSubdirs(subDirs []string) error {
 	}()
 
 	f.clearScanErrors(subDirs)
+	alreadyDeleted := make(map[string]struct{})
```

Comment:
To be honest, I am not sure if this is even needed. I guess there is no harm in handling the delete twice?

Reply:
It's not necessary because the delete detection is done on a new snapshot, i.e. already with all the changes from the scan including the newly added deletes, thus no repeat handling.